engine/src/main/java/org/terasology/world/block/tiles/WorldAtlasImpl.java

```diff
@@ -275,6 +303,23 @@ private void createTextureAtlas(final Texture texture) {
         return data;
     }
 
+    // Ref: http://stackoverflow.com/questions/221830/set-bufferedimage-alpha-mask-in-java/8058442#8058442
+    public void applyGrayscaleMaskToAlpha(BufferedImage image, BufferedImage mask) {
+        int width = image.getWidth();
+        int height = image.getHeight();
+
+        int[] imagePixels = image.getRGB(0, 0, width, height, null, 0, width);
+        int[] maskPixels = mask.getRGB(0, 0, width, height, null, 0, width);
+
+        for (int i = 0; i < imagePixels.length; i++) {
+            int color = imagePixels[i] & 0x00ffffff; // Mask preexisting alpha
+            int alpha = maskPixels[i] << 24; // Shift blue to alpha
+            imagePixels[i] = color | alpha;
+        }
+
+        image.setRGB(0, 0, width, height, imagePixels, 0, width);
+    }
+
```

Comment:
I understand, so the combination at this stage is purely for storage purposes prior to passing it to the GPU. 

I think perhaps the term applyGreyscaleMaskToAlpha is a bit misleading in the sense that it makes it sound like you have something in the alpha channel already and you are then applying/combining some further information onto it. In fact I seem to understand the alpha is empty and you are storing the glossyness map into that channel. I'd be more explicit and name the method "storeGreyscaleMapIntoAlpha()", but then use explicitly named parameters such as normalMap and glossinessMap when you call the method.